code/scripting/api/objs/object.cpp

```diff
@@ -594,5 +594,29 @@ ADE_FUNC(removeSound, l_Object, "soundentry GameSnd, [subsystem Subsys=nil]",
 	return ADE_RETURN_NIL;
 }
 
+
+ADE_FUNC(getIFFColor, l_Object, "number, number, number", 
+	"Gets the IFF color of the object",
+	"number, number, number", 
+	"IFF rgb color of the object or nil if object invalid")
+{
+	object_h* objh;
+
+	if (!ade_get_args(L, "o", l_Object.GetPtr(&objh)))
+		return ade_set_error(L, "iii", ADE_RETURN_NIL, ADE_RETURN_NIL, ADE_RETURN_NIL);
+
+	if (!objh->IsValid())
+		return ade_set_error(L, "iii", ADE_RETURN_NIL, ADE_RETURN_NIL, ADE_RETURN_NIL);
```

Comment:
Just return `ADE_RETURN_NIL` here.  Lua will apply it to any number of arguments the caller tries to get.  The way you have it here will return a valid triplet of `0, 0, 0`.

Reply:
Good to know, thanks! Just updated.